src/lib/comms/sol-oic-client.c

```diff
@@ -169,6 +169,21 @@ sol_oic_resource_ref(struct sol_oic_resource *r)
     return r;
 }
 
+static void
+clear_vector_list(struct sol_vector *vector, void *data)
+{
+    uint16_t i;
+    struct sol_str_slice *str;
+
+    if (data)
+        free(data);
+    else {
+        SOL_VECTOR_FOREACH_IDX(vector, str, i)
+            free((char *)str->data);
+    }
+    sol_vector_clear(vector);
```

Comment:
this looks like those `strv` (vector of strings) where the actual strings are allocated in a single blob, is this really the case? Usually one comes with that by getting a single string and splitting it, then if enough space (usually the delimiter), you `strdup()` the source string and replace delimiters with `\0`.

If that is the case, then you can avoid the `data` parameter and explicit storage in `r`, document the `[0]` is the memory and you can free in one go.